packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -46,6 +59,37 @@ export class ProblemWidget extends TreeWidget {
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
     }
 
+    @postConstruct()
+    protected init(): void {
+        super.init();
+        if (this.preferences.get('problems.autoReveal')) {
```

Comment:
I would extract a method `updateFollowActiveEditor`:
```ts
this.toDisposeOnCurrentWidgetChanged.dispose();
if (this.preferences.get('problems.autoReveal')) {
    this.toDisposeOnCurrentWidgetChanged.push(this.followActiveEditor());
    this.toDispose.push(this.toDisposeOnCurrentWidgetChanged);
}
```

and then do here:
```ts
this.updateFollowActiveEditor();
this.toDispose.push(this.preferences.onPreferenceChanged(e => {
    if (e.preferenceName ==== 'problems.autoReveal') {
        this.updateFollowActiveEditor();
    }
});
```

such approach takes care about disposing all listeners